JASP-Engine/JASP/R/correlation.R

```diff
@@ -15,806 +15,721 @@
 
-					width <- 580
-					height <- 580
+  results <- list()
 
-				} else if (l <= 2) {
+  meta <- list(
+      list(name="title", type="title"),
+    list(name="correlations", type="table"),
+    list(name="plot", type="image"))
 
-					width <- 400
-					height <- 400
+  results[[".meta"]] <- meta
 
-				} else {
-
-					width <- 250 * l
-					height <- 250 * l
+  results[["title"]] <- "Correlation Matrix"
+	results[["plot"]] <- correlation.plot
+	results[["correlations"]] <- correlation.table
+  
+	keep <- NULL
+	if (length(correlation.plot) > 0) {
+		keep <- correlation.plot$data
+  }
+      
+	if (perform == "init") {
 
-				}
+		if (length(options$variables) < 2) {
 
-				plot <- list()
+			return(list(results=results, status="complete", keep=keep))
 
-				plot[["title"]] <- "Correlation Plot"
-				plot[["width"]]  <- width
-				plot[["height"]] <- height
+		} else {
 
-				correlation.plot <- plot
-			}
+			return(list(results=results, status="inited", state=state, keep=keep))
 		}
-	}
-
-	if (perform == "run" && length(options$variables) > 0 && options$plotCorrelationMatrix) {
-
 
-		if (!is.null(state) && !is.null(diff) && ((is.logical(diff) && diff == FALSE) || (is.list(diff) && (diff$confidenceIntervals == FALSE && diff$confidenceIntervalsInterval == FALSE
-			&& diff$hypothesis == FALSE && diff$kendallsTauB == FALSE && diff$missingValues == FALSE && diff$pearson == FALSE && diff$plotCorrelationMatrix == FALSE
-			&& diff$plotDensities == FALSE && diff$plotStatistics == FALSE && diff$spearman == FALSE && diff$variables == FALSE)))) {
-
-			# if only "Report significance", "Flag significant correlations", "Means and Standard Deviations" or "Cross-product deviations and covariances" have changed, the previous plot can be used
+	} else { # run
+    
+    state <- list(options=options, correlationPlot=correlation.plot, tableData=tableData)
+    attr(state, "key") <- stateKey
+    
+		return(list(results=results, status="complete", state=state, keep=keep))
+	}
+}
 
-			correlation.plot <- state$correlationPlot
+.calculateCorrelations <- function(dataset, perform, options, stateTableData, variables=c(), pearson=TRUE, kendallsTauB=FALSE,
+	spearman=FALSE, hypothesis="correlated", CI=0.95) {
+  
+  ready <- FALSE
+  variables <- unlist(variables)
+  if (length(variables) == 0) {
+    variables <- c(variables, "...", "... ") # we need this trailing space so 1 != 2
+  } else if (length(variables) == 1) {
+    variables <- c(variables, "...")
+  } else {
+    ready <- TRUE
+  }
+  
+	tests <- c()
+	if (pearson)
+		tests <- c(tests, "pearson")
+	if (spearman)
+		tests <- c(tests, "spearman")
+	if (kendallsTauB)
+		tests <- c(tests, "kendall")
 
-		} else {
+  results <- list(tests=tests, variables=variables, hypothesis=hypothesis, CI=CI)
+  
+  if (length(tests) == 0) {
+    return(results)
+  }
+  
+  pairs <- combn(variables, 2, simplify=FALSE)
```

Comment:
are we okay not to add the name space for utils:: functions?

If so, ignore this. I forgot what I did

Reply:
Ya it seems kinda redundant; utils is a base package. Otherwise we'd also have to do base::c()